doc/metadata_routing.rst

```diff
@@ -179,42 +197,41 @@ consumers. In this example, we pass ``scoring_weight`` to the scorer, and
+(Meta-)Estimators which route metadata to other objects are :term:`routers <router>`.
+A(n) (meta-)estimator can be a :term:`consumer` and a :term:`router` at the same time.
+(Meta-)Estimators and splitters expose a `set_{method}_request` method for each method
+which accepts at least one metadata. For instance, if an estimator supports
+``sample_weight`` in ``fit`` and ``score``, it exposes
 ``estimator.set_fit_request(sample_weight=value)`` and
 ``estimator.set_score_request(sample_weight=value)``. Here ``value`` can be:
 
-- ``True``: method requests a ``sample_weight``. This means if the metadata is
-  provided, it will be used, otherwise no error is raised.
+- ``True``: method requests a ``sample_weight``. This means if the metadata is provided,
+  it will be used, otherwise no error is raised.
 - ``False``: method does not request a ``sample_weight``.
-- ``None``: router will raise an error if ``sample_weight`` is passed. This is
-  in almost all cases the default value when an object is instantiated and
-  ensures the user sets the metadata requests explicitly when a metadata is
-  passed. The only exception are ``Group*Fold`` splitters.
-- ``"param_name"``: if this estimator is used in a meta-estimator, the
-  meta-estimator should forward ``"param_name"`` as ``sample_weight`` to this
-  estimator. This means the mapping between the metadata required by the
-  object, e.g. ``sample_weight`` and what is provided by the user, e.g.
-  ``my_weights`` is done at the router level, and not by the object, e.g.
-  estimator, itself.
+- ``None``: router will raise an error if ``sample_weight`` is passed. This is in almost
+  all cases the default value when an object is instantiated and ensures the user sets
+  the metadata requests explicitly when a metadata is passed. The only exception are
+  ``Group*Fold`` splitters.
+- ``"param_name"``: alias for ``sample_weight`` if we want to pass different weights to
+  different consumers. If aliasing is used the meta-estimator should not forward
+  ``"param_name"`` to the consumer, but ``sample_weight`` instead, because the consumer
+  will expect a param called ``sample_weight``. This means the mapping between the
+  metadata required by the object, e.g. ``sample_weight`` and the variable name provided
+  by the user, e.g. ``my_weights`` is done at the router level, and not by the consuming
+  object itself.
 
 Metadata are requested in the same way for scorers using ``set_score_request``.
 
-If a metadata, e.g. ``sample_weight``, is passed by the user, the metadata
-request for all objects which potentially can consume ``sample_weight`` should
-be set by the user, otherwise an error is raised by the router object. For
-example, the following code raises an error, since it hasn't been explicitly
-specified whether ``sample_weight`` should be passed to the estimator's scorer
-or not::
+If a metadata, e.g. ``sample_weight``, is passed by the user, the metadata request for
```

Comment:
I've been wondering what word to use to refer to one of the entries in the dictionary that is passed as `params=`.

I think metadata is like data in that there is no plural. "The data show ...", "The author, title and page number are contained in the metadata". This means "a metadata" reads weird. How about "If an item of metadata, e.g. `sample weight`, is present in the metadata, the requests for all objects which can consume `sample_weight` should be configured by the user" (I also made some other changes to the sentence, just suggestions).

"an item of metadata" is the best I could come up with, I like it because in Python one entry in a dictionary (key and value) is already known as an item. This means the whole dictionary `{"sample_weight: ..., "groups": ..}` would be "the metadata" and one item of it (e.g. the weights) would be "an item of metadata".

You could argue that the weights by themselves are metadata, but then we need a term for the weights and groups together. Maybe "the set of metadata", but I like that less than my proposal above.


WDYT?

Reply:
What about "some metadata" (and then use plural)?

"set of" or "item of" both make me wonder what exactly is meant. And maybe we don't need to distinguish different quantities here.